src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts

```diff
@@ -187,21 +204,23 @@ export class TerminalSearchLinkOpener implements ITerminalLinkOpener {
 		return text;
 	}
 
-	private async _getExactMatch(sanitizedLink: string): Promise<URI | undefined> {
-		let exactResource: URI | undefined;
+	private async _getExactMatch(sanitizedLink: string): Promise<IResourceMatch | undefined> {
+		let resourceMatch: IResourceMatch | undefined;
 		if (osPathModule(this._os).isAbsolute(sanitizedLink)) {
 			const scheme = this._workbenchEnvironmentService.remoteAuthority ? Schemas.vscodeRemote : Schemas.file;
-			const resource = URI.from({ scheme, path: sanitizedLink });
+			const uri = URI.from({ scheme, path: sanitizedLink });
 			try {
-				const fileStat = await this._fileService.resolve(resource);
+				const fileStat = await this._fileService.resolve(uri);
 				if (fileStat.isFile) {
-					exactResource = resource;
+					resourceMatch = { uri, type: 'file'};
```

Comment:
Pass a boolean back instead of string? We use `isDirectory: boolean` in `resolvePath` so we should be consistent there. Could even move the interface into links.ts and share that with the resolvePath/cache code in TerminalLinkManager and the detectors as well